tests/flag_supported/test.c

```diff
@@ -0,0 +1,15 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+void useless(void)
+{
+    printf("I am not used\n");
+}
+
+const char main(int argc, char** argv)
```

Comment:
That's a funky looking `main` signature :)

Travis is complaining that the return value isn't `int` - but also isn't it more common to write `char **argv`? The `*` only applies to one argument, so that style avoids misleading declarations like `char** a, b`, where `a` and `b` look like they should both be `char **`, but `b` is just a `char`.

Reply:
The idea is to trigger a lot of warnings from the normal run of a compiler without the additional flags so that I can check that the right flags get added in order to get a clean build.
The funky main signature was intentional as a quick and easy way of generating some warnings in the newer compilers that started to care about how your `main` looks like.

Reply:
D'oh, sorry, I completely missed the point of the test.

I guess there are two issues with the travis runs then:
```
FAILED: build-in-src/target/objects/bob_test_flag_supported/flag_supported/test.c.o 

 gcc -c -Wall -Werror -Wno-discarded-qualifiers -Wno-ignored-qualifiers -Wno-main-return-type -MMD -MF build-in-src/target/objects/bob_test_flag_supported/flag_supported/test.c.o.d flag_supported/test.c -o build-in-src/target/objects/bob_test_flag_supported/flag_supported/test.c.o

flag_supported/test.c:9:12: error: return type of ‘main’ is not ‘int’ [-Werror=main]

 const char main(int argc, char** argv)

            ^

cc1: error: unrecognized command line option ‘-Wno-main-return-type’ [-Werror]

cc1: all warnings being treated as errors
```
- The return type error is caused by `-Werror=main` on this version, so do we need to try and add `add_if_supported -Wno-error=main` too?
- `Wno-main-return-type` is being added to the flags, even though it doesn't appear to be supported, which would suggest that there's a bug in the `add_if_supported` checking somewhere :(

Reply:
Found the two issues: old gcc uses '-Wmain' for warning about 'main' signature, so I've added '-Wno-main' to the flags. As for the invalid flags being added to the supported flags, it is due to the special way 'gcc' handles the '-Wno-<flag>' as command line parameter. If there is no actual file to compile 'gcc' will not warn about the parameter being unknown, but it will as soon as you pass a real file to compile. I've fixed it by stripping the 'no-' part from the flag and testing that with the compiler to see if it is supported.